core/src/main/scala/tofu/Context.scala

```diff
@@ -113,16 +113,24 @@ object Context {
     * object MyContext extends Context.Companion[MyContext]
     * }}}
     */
-  trait Companion[C] {
+  trait Companion[C] extends CompanionInstances[C] {
     type Has[F[_]] = WithContext[F, C]
 
-    implicit def tofuPromoteContextStructure[F[_], A](implicit
+    implicit def promoteContextStructure[F[_], A](implicit
```

Comment:
I think the priorities are wrong. `WithLocal` as a more concrete instance should be higher, and `WithContext` - lower.

Can you write a summon test?